website/blog/2020-05-25-7.10.0.md

```diff
@@ -0,0 +1,128 @@
+
+We also discovered that we still didn't support compiling a 5-years-old ECMAScript feature: `\u{...}`-style Unicode escapes! Thanks to [Justin](https://github.com/jridgewell), `@babel/preset-env` can now compile them by default both in strings and in identifiers.
+
+Lastly, we have added `@babel/plugin-proposal-class-properties` and `@babel/plugin-proposal-private-methods` to the [`shippedProposals`](https://babeljs.io/docs/en/babel-preset-env#shippedproposals) option of `@babel/preset-env`. These proposals are not Stage 4 (i.e. part of the ECMAScript standard) yet, but they are already enabled by default in [many JavaScript engines](https://github.com/tc39/proposal-class-fields#implementations).
+
+### Improved optional chaining `?.` ergonomics ([#10961](https://github.com/babel/babel/pull/10961), [#11248](https://github.com/babel/babel/pull/11248))
+
+In TypeScript 3.9, the interaction between non-null assertions (postfix `!`) and optional chaining [has been changed](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-9.html#parsing-differences-in-optional-chaining-and-non-null-assertions), to make it more useful.
+
+```ts
+foo?.bar!.baz
+```
+
+In TypeScript 3.8 and Babel 7.9, that could would be read as `(foo?.bar)!.baz`: "If `foo` is not nullish, get `.bar` from it. Then trust me that `foo?.bar` is never nullish and always get `.bar` from it". This means that when `foo` is nullish that code would always throw, because we are trying to get `.baz` from `undefined`.
+
+In TypeScript 3.9 and Babel 7.10, that code behaves similarly to `foo?.bar.baz`: "If `foo` is not nullish, get `.bar.baz` from it and trust me that `foo?.bar` isn't nullish".
+
+Additionally, the class fields proposal [recently supports](https://github.com/tc39/proposal-class-fields/pull/301) mixing the optional chaining `?.` with private fields. This means that this code is now valid:
+
+```js
+obj?.property.#priv;
+obj?.#priv;
+```
+
+Note that in the second example, if `obj` is not nullish and it doesn't have the `#priv` field, it would still throw an error (exactly as `obj.#priv` would throw). You can read the next paragraph to see how to avoid it!
+
+### Private Fields in `in` ([#11372](https://github.com/babel/babel/pull/11372))
+
+```js
+class Person {
+  #name;
+  
+  hug(other) {
+    if (#name in other) console.log(`${this.#name} 🤗 {other.#name}`);
+    else console.log("It's not a person!")
+  }
+}
+```
+
+This stage 1 proposal allows to statically check if a given object has a specific private field.
+
+Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. We can often leverage this characteristic to use a `try`/`catch` statement and detect their absence, but this proposal gives us a more compact and robust syntax to do so.
```

Comment:
```suggestion
Private fields have a built-in "brand check": if you try to access them in an object where they aren't defined, it will throw an exception. You can detect undefined private fields by leveraging this behavior with a `try`/`catch` statement, but this proposal gives us a more compact and robust syntax to do so.
```